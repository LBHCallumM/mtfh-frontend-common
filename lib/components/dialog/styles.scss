@import "node_modules/lbh-frontend/lbh/base";
@import "node_modules/lbh-frontend/lbh/components/lbh-dialog/dialog";

[data-reach-dialog-overlay] {
  z-index: 100;
}

.lbh-dialog__close {
  margin-top: 1.5rem;
  margin-right: 1.5rem;
}

.lbh-dialog__title {
  margin-right: 2rem;
}
