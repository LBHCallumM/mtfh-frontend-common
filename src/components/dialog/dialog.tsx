import React, { ComponentPropsWithoutRef, forwardRef, useState } from 'react';
import Dialog from 'lbh-frontend/lbh/components/lbh-dialog';

import './styles.scss';

export interface CustomDialogProps extends ComponentPropsWithoutRef<'dialog'> {
  title?: string;
}

export const CustomDialog = forwardRef<HTMLTextAreaElement, any>(
  function CustomDialog({ title, ...props }) {
    const [open, setOpen] = useState(false);

    return (
      <>
        <Dialog
          title="Are you sure?"
          isOpen={open}
          onDismiss={() => setOpen(false)}
        >
          {props}
        </Dialog>
      </>
    );
  }
);
