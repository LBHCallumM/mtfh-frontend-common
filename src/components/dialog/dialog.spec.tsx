import React from 'react';
import { render } from '@testing-library/react';

import { CustomDialog } from './dialog';

test('it renders correctly', async () => {
  const { container } = render(
    <CustomDialog title="test">
      <h1>Test</h1>
      <button>Test Button</button>
    </CustomDialog>
  );
  expect(container).toMatchSnapshot();
});
