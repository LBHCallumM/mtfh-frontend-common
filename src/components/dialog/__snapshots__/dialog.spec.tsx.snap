// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <button
    class="govuk-button lbh-button"
  >
    Preview dialog
  </button>
</div>
`;
